### PR TITLE
fix: replace old design templates with new design system

### LINKED ISSUE #9699: Remove unused old design templates and remain the newer templates
Implementation follow-up issue for https://github.com/GoogleChrome/web.dev/issues/9650.

**List of the unused Nunjucks files**

- /_includes/partials/footer.njk - need to be replaced with /_includes/partials/site-footer.njk 
- /_includes/partials/header-default.njk - need to be replaced with /_includes/partials/site-header.njk 
- /_includes/default.njk - need to be replaced with /_includes/default-next.njk 

**Note:** Since the files with -next suffixed will be removed, all references which direct to `-next.njk` should be updated to the correct file name as well.

### PR DESCRIPTION
**Pull request by mamieorine**
Fixes #9699 

Changes proposed in this pull request:

- Remove the old design templates and remain the new design system templates
   - Removes /_includes/partials/footer.njk template and remains /_includes/partials/site-footer.njk.
   - Replaces code in /_includes/partials/header-default.njk template with /_includes/partials/site-header.njk and change all `header-default.njk` references to `site-header.njk`
   - Replaces code in /_includes/default.njk template with /_includes/default-next.njk and change all `default-next.njk` references to `default.njk`

### CODE REVIEW

**Review by jimjiminyjimjim — COMMENTED** [review]
lgtm

**Review by matthiasrohmer — COMMENTED** [review]
Ran a diff build, no changes which is good 🙂  One question though and one fix we could do on the way.

**Comment by matthiasrohmer** [comment]
... and we got a bunch of conflicts now that I merged some of your other PRs, which need to be resolved.

**Comment by mamieorine** [comment]
@matthiasrohmer  This PR is ready for re-review, thank you 😊

File: src/site/_includes/partials/head.njk [hunk 1]
```diff
@@ -1,23 +1,10 @@
...
-  {% else %}
-    {# hashForProd() will cache bust the css in dev with a random string. #}
-    <link rel="stylesheet" href="{{ helpers.hashForProd('/css/main.css') }}">
-  {% endif %}
-
-  <link rel="preload" as="font" crossorigin="anonymous" href="/fonts/material-icons/regular.woff2">
-{% endif %}
+{% include "partials/theme-script.njk" %}
+<link rel="stylesheet" href="{{ helpers.hashForProd('/css/next.css') }}">
+<link rel="stylesheet" href="{{ helpers.hashForProd('/css/legacy-rollout.css') }}">
```
Comment (matthiasrohmer) [inline src/site/_includes/partials/head.njk:6]: This is removed in https://github.com/GoogleChrome/web.dev/pull/9781, right? So we can probably remove the link here already, no?

File: src/site/_includes/partials/head.njk [hunk 2]
```diff
@@ -1,23 +1,10 @@
...
-{% if CSS_ORIGIN === 'next' %}
-  {% include "partials/theme-script.njk" %}
-  {# hashForProd() will cache bust the css in dev with a random string. #}
-  <link rel="stylesheet" href="{{ helpers.hashForProd('/css/next.css') }}">
-  <link rel="stylesheet" href="{{ helpers.hashForProd('/css/legacy-rollout.css') }}">
-  <link rel="preload" as="font" crossorigin="anonymous" href="/fonts/material-icons/regular.woff2">
-{% else %}
-  {% if process.env.ELEVENTY_ENV === 'prod' %}
-    {# CSS will be injected by purify-css transform. #}
-    <!-- __PURGECSS_INJECT -->
```
Comment (matthiasrohmer) [inline src/site/_includes/partials/head.njk:13]: As next.css is not inlined we can actually remove all the CSS purging logic as well: https://github.com/GoogleChrome/web.dev/blob/main/src/site/_transforms/purify-css.js this also needs to be removed in the 11ty config then.

**Review by matthiasrohmer — APPROVED** [review]

**Pull request merged**